vm/string.go

```diff
@@ -471,6 +472,719 @@ func builtinStringInstanceMethods() []*BuiltInMethodObject {
 				}
 			},
 		},
+		{
+			// Returns a string which is concatenate with the input string or character
+			//
+			// ```ruby
+			// "Hello ".concat("World") # => "Hello World"
+			// ```
+			// @return [String]
+			Name: "concat",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+
+					str := receiver.(*StringObject).Value
+					concatStr, ok := args[0].(*StringObject)
+
+					if !ok {
+						return wrongTypeError(receiver.Class())
+					}
+
+					return t.vm.initStringObject(str + concatStr.Value)
+				}
+			},
+		},
+		{
+			// Returns the character of the string with specified index
+			// It will raise error if the input is not an Integer type
+			//
+			// ```ruby
+			// "Hello"[1]        # => "e"
+			// "Hello"[5]        # => nil
+			// "Hello\nWorld"[5] # => "\n"
+			// "Hello"[-1]       # => "o"
+			// "Hello"[-6]       # => nil
+			// ```
+			//
+			// @return [String]
+			Name: "[]",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+					if len(args) != 1 {
+						return initErrorObject(ArgumentErrorClass, "Expect 1 argument. got=%v", strconv.Itoa(len(args)))
+					}
+
+					str := receiver.(*StringObject).Value
+					i := args[0]
+					index, ok := i.(*IntegerObject)
+
+					if !ok {
+						return initErrorObject(TypeErrorClass, "Expect Integer. got=%T (%+v)", args[0], args[0])
+					}
+
+					indexValue := index.Value
+
+					if indexValue < 0 {
+						if -indexValue > len(str) {
+							return NULL
+						}
+						return t.vm.initStringObject(string([]rune(str)[len(str)+indexValue]))
```

Comment:
Reference:
https://stackoverflow.com/questions/15018545/how-to-index-characters-in-a-golang-string

This case if for `ASCII` only:
```
string[len(str)+indexValue]
```

However, the transition to `rune` type can be used to `UTF-8` condition, try the example below:
```
package main

import "fmt"

func main() {
    fmt.Println(string("Hello, 世界"[1]))         // ASCII only
    fmt.Println(string("Hello, 世界"[8]))         // ASCII only
    fmt.Println(string([]rune("Hello, 世界")[1])) // UTF-8
    fmt.Println(string([]rune("Hello, 世界")[8])) // UTF-8
}
```

Got:
```
e
¸
e
界
```